### ISSUE TITLE
UI: Audio Mixer colors are hard to distinguish

### ISSUE THREAD

**Issue by VazMike** [obsproject/obs-studio#4604]
### Operating System Info

macOS 11.2

### OBS Studio Version

26.1.2

### OBS Studio Log URL

Not neessary.

### Expected Behavior

I expect a higher contrast between green and yellow and sharper dB units.

### Current Behavior

On default OBS theme green and yellow zones are hard to distinguish.
I admin I'm a bit color blind. But it is mild. There are many people whose condition is worse.
On OBS Project tutorial page colors look fine:

![](https://i.imgur.com/eCYfThu.png)

But in my app they look:

<img width="560" alt="image" src="https://user-images.githubusercontent.com/14861435/116250757-afa90000-a776-11eb-9d02-ddaea7486cbb.png">

I'm using latest MacBook Air display with P3 gamut.

BTW the dB units look like stretched out low-resolution image. The text is not sharp. 

### Steps to Reproduce

Open Audio Mixer.

### Anything else we should know?

I admit this is not a technical issue, but it maybe considered as a UX issue.

**Comment by Fenrirthviti** [comment]
Just for reference, the theme used in the tutorial page is Rachni, so you can switch to that (Settings -> General) if that is a sufficient solution for you? The default theme uses "industry standard" colors as far as I'm aware.

**Comment by VazMike** [comment]
*Rachni* theme indeed has a better color contrast. But that doesn't mean *Default Dark* theme shouldn't be fixed.
And blurry dB units are present in both themes.

**Comment by Fenrirthviti** [comment]
What would you propose as the new color codes? The contrast, frankly, looks just fine to my eyes.

**Comment by VazMike** [comment]
I don't know, people are color blind differently. What will work better for me might not work well for others.
To look at it objectively you can increase the color contrast which is the numerical value.
Current hex values are `#417D33` for green and `#7E7F36` for orange, the contrast is `1.18` which is very low.
For reference contrast between white and black is `21` and contrast between 2 same colors is `1`.

The idea has just come up to my head. Maybe it would beneficial to use subtle separators between color zones?

**Comment by ElectronicWar** [comment]
Most other softwares or games have special colour blind modes, so maybe a new theme for red-green blindness could be added instead of modifying the default themes?

**Comment by VodBox** [comment]
I will be looking into alternative default mixer colors that should hopefully not significantly change their appearance, but should hopefully make them more distinct. The issue is that green, yellow and red for mixer colors is a standard that people working with audio expect, so there may be only so much that can be done for certain forms of color-blindness.

At the very least down the line I'll be adding an explicit accessibility option to override the mixer colors from the current theme with known color combinations that are good for a given form of color-blindness, so even if our default doesn't see significant improvement, there will be options available.

**Comment by VazMike** [comment]
@VodBox Thank you, this is a clean solution.

And since you are already working on audio mixer UI, could you also make numbers sharper?

**Comment by VodBox** [comment]
> And since you are already working on audio mixer UI, could you also make numbers sharper?

I believe @WizardCM has already looked into this, and it has to do with DPI scaling in our UI framework. I can certainly take a look at it myself, but that is a separate issue to this one. I believe the relevant issue for that is this: [https://github.com/obsproject/obs-studio/issues/3915](https://github.com/obsproject/obs-studio/issues/3915)

**Comment by RytoEX** [comment]
As an aside, the volume meter colors are defined in the theme stylesheet files in the OBS Studio application/installation folder (`data\obs-studio\themes`, search for "VolumeMeter" in the theme file), which are CSS-like syntax.  You can edit the color codes there to colors that work best for you as a workaround.  See our wiki page on [Custom Themes](https://obsproject.com/wiki/Custom-Themes) for more info.

As @VodBox said, we'd like to work towards and provide a more comprehensive solution so that users can easily choose what's best for their individual needs without having to manually edit theme files.  It's definitely on our radar.

> And blurry dB units are present in both themes.

@VodBox already said this as well: #3915 already exists regarding the legibility of the text in the volume meter.  The issue with the blurry text has to do with QPainter (or some other part of Qt) and High DPI.  We'll keep this issue scoped on color contrast issues with the meter colors, and the other issue scoped on blurry text in the volume meter.

**Comment by RytoEX** [comment]
This should be addressed by #4975, which has been merged. Audio Mixer colors can now be set to custom colors from within OBS in the new Accessibility Settings in the OBS Studio 28 beta and release candidate builds, which are available now.
